### PR TITLE
Altitude Angel updates January 2022

### PR DESCRIPTION
**Pull request by rupertbenbrook**
Major updates to the Altitude Angel integration features including:
- Converting extension library to .NET Standard 2.0, and isolating all .NET Framework-based components in the main Mission Planner project
- Threading and async improvements improving stability and performance.
- Map overlay configuration hierarchical layers switches, varying overlay transparency, and which view have the overlays.
- Map overlay details are now displayed when clicked in a docked flyout panel, improving the detail of the information provided, and providing a better map navigation experience.
- Major updates to flight reporting, allowing greater control of flight plan attributes.
- Updates to login to allow customers with their own Altitude Angel subscriptions to use their paid for services in Mission Planner.
- Integration of Altitude Angel tactical deconfliction and telemetry APIs, which provides drone position reporting and deconfliction guidance and instructions.

### CODE REVIEW

**Comment by meee1** [comment]
so first issue i have come across is, i currently have 5 windows open, and they just keep opening.

the getdata async operation is causing the signin page to keep reapearing.

>	MissionPlanner.exe!MissionPlanner.Utilities.AltitudeAngel.WpfAuthorizeDisplay.GetCodeUri(System.Uri authorizeUri, System.Uri redirectUri) Line 28	C#
 	AltitudeAngelWings.dll!AltitudeAngelWings.ApiClient.Client.UserAuthenticationTokenProvider.GetToken(System.Threading.CancellationToken cancellationToken) Line 38	C#
 	AltitudeAngelWings.dll!AltitudeAngelWings.ApiClient.Client.BearerTokenHttpMessageHandler.SendAsync(System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) Line 26	C#
 	[External Code]	
 	AltitudeAngelWings.dll!AltitudeAngelWings.ApiClient.Client.AltitudeAngelClient.GetMapData.AnonymousMethod__0() Line 67	C#
 	[External Code]	
 	AltitudeAngelWings.dll!AltitudeAngelWings.ApiClient.Client.AltitudeAngelClient.GetMapData(AltitudeAngelWings.ApiClient.Models.BoundingLatLong latLongBounds, System.Threading.CancellationToken cancellationToken) Line 67	C#
 	AltitudeAngelWings.dll!AltitudeAngelWings.Service.AltitudeAngelService.UpdateMapData(AltitudeAngelWings.Extra.IMap map, System.Threading.CancellationToken cancellationToken) Line 129	C#
 	AltitudeAngelWings.dll!AltitudeAngelWings.Service.AltitudeAngelService..ctor.AnonymousMethod__21_12(System.Threading.CancellationToken ct) Line 59	C#
 	[External Code]

**Comment by meee1** [comment]
another issue - deadlocks

image

**Comment by meee1** [comment]
image
another

**Comment by rupertbenbrook** [comment]
Thanks @meee1, I'll take a look at these. I think some could be to do with the change in format of our authentication tokens from opaque tokens to JWT, so might be because your local config already had old token stored. Could you let me know what version you had running prior to running these tests so I can test the migration of the token from one version to another?

**Comment by meee1** [comment]
i had started fresh, as i needed to enter credentials to even get that far

**Comment by rupertbenbrook** [comment]
OK, I've updated the branch correctly this time. Apologies for the mistake. Let me know how you get on with this one - I'm hoping the problems above are addressed.

**Comment by rupertbenbrook** [comment]
Hi @meee1, I've merged the upstream changes into the PR to fix the merge conflicts.

**Comment by rupertbenbrook** [comment]
Hi @meee1, have you had a change to look at the updated changes?

**Comment by meee1** [comment]
can i request a rebase on this? as there are merge conflicts, and the history is getting messy with the merges

**Comment by rupertbenbrook** [comment]
Sure, no problem.

**Comment by meee1** [comment]
There are still merge conflicts I can't merge it like this

**Comment by rupertbenbrook** [comment]
I've not finished just yet, was just merging before rebasing. I'm on a bit of a poor internet connection at the moment, so pulling everything has taken a while.

**Comment by rupertbenbrook** [comment]
OK, rebase merge complete. How's that?

File: app.config [hunk 1]
```diff
@@ -7,24 +7,29 @@
     <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.7.2"/>
   </startup>
   <appSettings>
-    <add key="UpdateLocationVersion" value="https://firmware.oborne.me/MissionPlanner/upgrade/version.txt"/>
-    <add key="UpdateLocation" value="https://firmware.ardupilot.org/MissionPlanner/upgrade/"/>
-    <add key="UpdateLocationMD5" value="https://firmware.ardupilot.org/MissionPlanner/checksums.txt"/>
-    <add key="UpdateLocationZip" value="https://firmware.ardupilot.org/MissionPlanner/MissionPlanner-latest.zip"/>
-    <add key="BetaUpdateLocationVersion" value="https://github.com/ArduPilot/MissionPlanner/releases/download/betarelease/version.txt"/>
```
Comment (meee1) [inline app.config:14]: the issue is is stuff like this is a regression, and shouldnt be in the pull request

Comment (rupertbenbrook) [reply app.config:14]: OK, perhaps I'm best off completely starting again with a new pull request forked anew from master. I'll put that together in parallel to this one.

**Pull request closed without merging**